File: app/src/main/java/tk/lorddarthart/accurateweathertestapp/TaskLoader.java
package tk.lorddarthart.accurateweathertestapp;

import android.os.AsyncTask;

public final class TaskLoader {

    private static AsyncTask task;

    private TaskLoader() {
        throw new UnsupportedOperationException();
    }

    public static void setTask(AsyncTask task) {
        TaskLoader.task = task;
    }

    public static void cancel() {
        TaskLoader.task.cancel(true);
    }
}

File: app/src/main/java/tk/lorddarthart/accurateweathertestapp/util/TaskLoader.kt
package tk.lorddarthart.accurateweathertestapp.util

import android.os.AsyncTask

class TaskLoader private constructor() {

    init {
        throw UnsupportedOperationException()
    }

    companion object {

        private var task: AsyncTask<*, *, *>? = null

        fun setTask(task: AsyncTask<*, *, *>) {
            Companion.task = task
        }

        fun cancel() {
            task!!.cancel(true)
        }
    }
}
